app/javascript/src/components/Editor.tsx

```diff
@@ -72,6 +72,8 @@ class Editor extends React.Component<EditorProps, EditorState> {
     if (JSON.stringify(story) !== JSON.stringify(newStory)) {
       this.updateStory(newStory)
       this.forceUpdate()
+    } else {
+      console.log('nothing new')
```

Comment:
this seems a little out of place in this PR. Possible that you were mixing up your branches while working on url hashes + undo/redo?

Reply:
Yeah I accidentally forgot to switch branches from `undo-redo` when initially working on this. I stashed the work, reset `undo-redo` and checked out `url-hash-progress` from `master` and then applied the stash. Unfortunately some of the history tracking logic in `Editor` got carried over (I must have messed something up) so I deleted and added that line. I'll try resetting just `Editor` so there are no changes pushed on this branch

Reply:
Gotchya, yeah a couple tips I've got for you which might help you keep organized around that stuff a little more:
- make the branch name visible in your terminal (or somewhere prominent), which will make it easier for you to have a sense for what branch you're on. Which will increase the likelihood that you keep your work more isolated / organized
  - if you're on vanilla bash, here's a bit of .bash_profile goodness: 
    ```bash
    display_git_branch() {
      branch=$(git branch 2> /dev/null | sed -e '/^[^*]/d' -e 's/* \(.*\)/\1/')
      if [ "$branch" != "" ]; then
        echo "($branch)"
      fi
    }
    export PS1="\[\e[1;31m\]BASH:\W\$(display_git_branch) $ \[\e[0m\]"
    ```
- use [gitx](http://rowanj.github.io/gitx/) when assembling together commits. I'm normally not a fan of visual interfaces for dealing with programming stuff, except for when it comes to gitx.